src/DenseACSets.jl

```diff
@@ -293,13 +293,22 @@ function ACSetInterface.subpart_type(x::DynamicACSet,s::Symbol)
 end
 
 """Cast StructACSet into a DynamicACSet"""
-function DynamicACSet(X::StructACSet{S}) where S 
-  Y = DynamicACSet(string(typeof(X).name.name), Schema(S); type_assignment=datatypes(X))
+function DynamicACSet(X::StructACSet{S, Ts, PT}) where {S, Ts, PT} 
+  Y = DynamicACSet(string(typeof(X).name.name), Schema(S);
+        type_assignment=datatypes(X), index = indices(X), 
+        unique_index = unique_indices(X), part_type=PT)
   copy_parts!(Y,X, NamedTuple(Dict(k=>parts(X,k) for k in types(S))))
```

Comment:
I think this line can be simplified to `copy_parts!(Y,X)`.

Reply:
Right now this doesn't work, because dispatch goes to 

```
ACSetInterface.copy_parts!(to::SimpleACSet, from::SimpleACSet; kw...) = copy_parts!(to, from, (;kw...))
```

When you have mixing between StructACSets and DynamicACSets. That method doesn't call `common_objects` by default (it's meant to be used like `copy_parts!(G, H; V=[1,3], E=[...])`.

I've removed the `ACSetInterface.copy_parts!(to::DynamicACSets, from::DynamicACSets)` method and made it a special case of an updated `ACSetInterface.copy_parts!(to::SimpleACSet, from::SimpleACSet)` method, which uses `common_objects` iff the kwargs are empty. Let me know what you think of that!

Reply:
Great. It didn't even occur to me that this would be a problem. In general, I think that static and dynamic acsets should interact as seamlessly as possible, so glad to see more uniform behavior for `copy_parts!`.